protocols/imap/src/main/java/org/apache/james/imap/processor/base/UidMsnConverter.java

```diff
@@ -52,8 +52,8 @@ public NullableMessageSequenceNumber getMsn(MessageUid uid) {
 
     public Optional<MessageUid> getUid(int msn) {
         return uids
-            .take(msn)
-            .lastOption()
```

Comment:
Can you explain the difference?

Reply:
> Can you explain the difference?

given 3 elements [1, 2, 3]

take 50 => [1, 2, 3] => last => 3
drop 50 => [] => head => empty